src/query/function_score_query.rs

```diff
@@ -0,0 +1,285 @@
+use std::fmt;
+use std::sync::Arc;
+
+use crate::query::{EnableScoring, Explanation, Query, Scorer, Weight};
+use crate::{DocId, DocSet, Score, SegmentReader, TantivyError};
+
+type Function = Arc<dyn Fn(&SegmentReader, Score, DocId) -> Score + Send + Sync + 'static>;
```

Comment:
I don't like this...

If your fucntion needs to access a column from a fast field, you don't want to access it at each call. 

The function should be a factory that takes a `SegmentReader` and returns a scoring function